client.lua

```diff
@@ -89,11 +91,64 @@ local function hasHarness(items)
     harness = _harness
 end
 
+-- Weapon Stress [Haroki]
+local function IsWhitelistedWeaponStress(weapon)
+    if weapon then
+        for _, v in pairs(config.WhitelistedWeaponStress) do
+            if weapon == v then
+                return true
+            end
+        end
+    end
+    return false
+end
+
+local hasWeapon = false
+
+local function startWeaponStressThread(weapon)
+    if IsWhitelistedWeaponStress(weapon) then hasWeapon = false; return end
+
+    CreateThread(function()
+        while hasWeapon do
+            local ped = cache.ped
+            if IsPedShooting(ped) then
+                if math.random() < config.StressChance then
+                    TriggerServerEvent('hud:server:GainStress', math.random(1, 3))
+                end
+            end
+            Wait(0)
+        end
+    end)
+end
+
+AddEventHandler('ox_inventory:currentWeapon', function(currentWeapon)
+    if not currentWeapon then
+        hasWeapon = false
+        return
+    end
+
+    local curGun = currentWeapon.name
+    hasWeapon = true
+    startWeaponStressThread(curGun)
+end)
+
+-- Weapon Stress end
+AddEventHandler('onResourceStart', function (resource)
+    if not resource == GetCurrentResourceName() or not LocalPlayer.state.isLoggedIn then return end
+    local Weapon = OxInventory:getCurrentWeapon()
+    PlayerData = QBCore.Functions.GetPlayerData()
+    PlayerLoaded = true
+    if Weapon then startWeaponStressThread(Weapon.name) end
+end)
+
+
 RegisterNetEvent("QBCore:Client:OnPlayerLoaded", function()
     Wait(2000)
     local hudSettings = GetResourceKvpString('hudSettings')
+    local Weapon = OxInventory:getCurrentWeapon()
     if hudSettings then loadSettings(json.decode(hudSettings)) end
     PlayerData = QBCore.Functions.GetPlayerData()
+    if Weapon then startWeaponStressThread(Weapon.name) end
```

Comment:
```suggestion
    if hudSettings then loadSettings(json.decode(hudSettings)) end
    PlayerData = QBCore.Functions.GetPlayerData()
```
This check doesn't really belong here (will already be handled by the `ox_inventory:currentWeapon` event).